llmfoundry/command_utils/train.py

```diff
@@ -187,6 +188,24 @@ def _initialize_dist_with_barrier(dist_timeout: Union[int, float]):
     log.debug('Barrier test passed with device.')
 
 
+def _sort_callbacks(trainer: Trainer):
+    """Sort callback so that checkpoint saving callbacks go first.
+
+    Args:
+        trainer (Trainer): Trainer object
+    """
+
+    def _sort_key(c: Callback) -> int:
+        # CheckpointSaver goes first because the blocking time is shortest while upload is async.
+        if isinstance(c, CheckpointSaver):
+            return 0
+        if isinstance(c, HuggingFaceCheckpointer):
+            return 1
+        return 2
```

Comment:
should other callbacks be before or after ckpting? imo before as they might change model

Reply:
Ah good point, i'll switch to before